### PR TITLE
Chore/move grid into main repo

### LINKED ISSUE #7079: Linked Foreign Key Icon barely visible -> resizing problem
# Bug report

## Describe the bug

When in table view and a column has a foreign key then it has a specific "linked" icon.
HOWEVER this icon is barely visible due to its sizing technology (see video attached).

## To Reproduce

1. Go to any table with foreign key
2. Make the column with foreign key as small as possible
3. See that the icon was scaled down to nearly 0

## Expected behavior

Icon should be clearly visible all time.

## Screenshots

https://user-images.githubusercontent.com/9058762/137900591-a932295e-4fa4-4813-a3f9-b7a91f780b3c.mp4

## System information

- OS: macOS Latest
- Brave Browser

### PR DESCRIPTION
**Pull request by saltcod**
## What kind of change does this PR introduce?

Moves supabase/grid into the main repo

https://github.com/supabase/grid

### CODE REVIEW

File: studio/components/grid/query/Query.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+interface IQuery {
+  from: (table: string, schema?: string) => IQueryAction;
+}
+
+export class Query implements IQuery {
+  /**
+   * @param name    the table name.
+   * @param schema  the table schema, by default set to 'public'.
+   */
+  from(name: string, schema?: string) {
```
Comment (saltcod) [inline studio/components/grid/query/Query.ts:12]: @phamhieu this is giving me a ts error. Wondering if you've got any ideas.

`Property 'from' in type 'Query' is not assignable to the same property in base type 'IQuery'.`

Comment (phamhieu) [reply studio/components/grid/query/Query.ts:12]: Need to fix QueryAction.insert values param to `Dictionary<any> | Dictionary<any>[]` as the interface

File: studio/components/grid/query/QueryAction.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    return new QueryFilter(this.table, 'delete', undefined, options);
+  }
+
+  /**
+   * Performs an INSERT into the table.
+   *
+   * @param values             The values to insert.
+   * @param options.returning  If `true`, return the inserted row(s) in the response.
+   */
+  insert(values: Dictionary<any>[], options?: { returning: boolean }) {
```
Comment (phamhieu) [inline studio/components/grid/query/QueryAction.ts:43]:
```suggestion
  insert(values: Dictionary<any> | Dictionary<any>[], options?: { returning: boolean }) {
```

File: studio/components/grid/query/QueryAction.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import { Dictionary, QueryTable } from '../types';
+import { IQueryFilter, QueryFilter } from './QueryFilter';
+
+export interface IQueryAction {
+  count: () => IQueryFilter;
+  delete: (options?: { returning: boolean }) => IQueryFilter;
+  insert: (
+    values: Dictionary<any> | Dictionary<any>[],
```
Comment (phamhieu) [inline studio/components/grid/query/QueryAction.ts:8]:
```suggestion
    values: Dictionary<any>[],
```

Comment (saltcod) [reply studio/components/grid/query/QueryAction.ts]: Perfect. Thanks Hieu.

**Comment by saltcod** [comment]
@MildTomato got a couple of small css discrepancies between the old repo and the new that I'm not sure how to resolve.  I removed the old tailwind config has explicit font sizes, while the one in studio doesn't. 

https://github.com/supabase/grid/blob/main/tailwind.config.js#L66

Do we just use the default Tailwind font sizes in studio? If so, we might need to style these grid text values explicitly?


CleanShot 2022-05-27 at 15 29 14@2x

**Comment by saltcod** [comment]
@phamhieu @joshenlim @MildTomato any idea why the build is failing on this?   Builds clean locally.

**Review by MildTomato — APPROVED** [review]

**Comment by MildTomato** [comment]
changed content of text to be correct size

File: studio/components/grid/query/QueryAction.ts (hunk 3 above)
Comment (MildTomato) [reply studio/components/grid/query/QueryAction.ts]: Resolved?

**Comment by MildTomato** [comment]
We've removed the package (tiny-popover) that was causing issues. using our own popover from @supabase/ui now.

**Pull request merged**